pdal_ign_macro/macro.py

```diff
@@ -83,6 +87,40 @@ def keep_non_planar_pts(pipeline, condition, condition_out):
     return pipeline
 
 
+def remove_dimensions(input_las, dimensions, ouput_las):
+    """
+    export new las whitout some dimensions
+    """
+    pipeline = pdal.Pipeline() | pdal.Reader.las(input_las)
+    pipeline.execute()
+    points = pipeline.arrays[0]
+
+    dim_remove = []
+    for dim in dimensions:
+        dim_remove.append(list(points.dtype.fields.keys()).index(dim))
+
+    new_points = []
+    for pt in points:
+        ptl = list(pt)
+        ptl = [ptl[i] for i in range(len(ptl)) if i not in dim_remove]
+        new_points.append(tuple(ptl))
+
+    new_dtype = {}
+    for dim in points.dtype.fields.keys():
+        if dim in dimensions:
+            continue
+        new_dtype[dim] = points.dtype.fields[dim]
+
+    new_las = numpy.array(new_points, dtype=new_dtype)
+    params = li.get_writer_parameters_from_reader_metadata(pipeline.metadata)
+
+    pipeline_end = pdal.Pipeline(arrays=[new_las], )
```

Comment:
Ca peut se simplifier en utilisant numpy : 
```suggestion
    pipeline.execute()
    points = pipeline.arrays[0]

    input_dimensions = list(points.dtype.fields.keys())
    output_dimensions = [dim for dim in input_dimensions if dim not in dimensions]

    points_pruned = points[output_dimensions]

    params = li.get_writer_parameters_from_reader_metadata(pipeline.metadata)

    pipeline_end = pdal.Pipeline(
        arrays=[points_pruned],
    )
```

Reply:
vu comme ça... il faut tout de même ajouter le dtype au tableau.

Reply:
pour moi c'est numpy qui s'en occupe tout seul, parce que la donnée est une structured array https://numpy.org/doc/stable/user/basics.rec.html , mais si t'as un doute ça vaut le coup de vérifier

Reply:
oui tu as raison. Je me suis vraiment complexifié la vie !! il va falloir que tu me donnes des cours de numpy